File: tests/test_stats.py
import unittest
from src.stats import mean, median


class TestStats(unittest.TestCase):

    def test_mean(self):
        self.assertEqual(mean([1.0, 2.0, 3.0, 4.0, 5.0]), 3.0)
        self.assertIsNone(mean([]))
        self.assertEqual(mean([7.0]), 7.0)
        self.assertEqual(mean([-1.0, -2.0, -3.0, -4.0, -5.0]), -3.0)

    def test_median(self):
        self.assertEqual(median([1.0, 2.0, 3.0]), 2.0)
        self.assertEqual(median([1.0, 2.0, 3.0, 4.0]), 2.5)
        self.assertIsNone(median([]))
        self.assertEqual(median([7.0]), 7.0)


if __name__ == "__main__":
    unittest.main()

File: tests/test_stats.rs
#[cfg(test)]
mod tests {
    use yijia_ids706_mini_proj7::stats::{mean, median};

    #[test]
    fn test_mean() {
        // Test with a normal list of numbers
        let numbers = vec![1.0, 2.0, 3.0, 4.0, 5.0];
        assert_eq!(mean(&numbers), Some(3.0));

        // Test with an empty list
        let numbers: Vec<f64> = vec![];
        assert_eq!(mean(&numbers), None);

        // Test with a list of one element
        let numbers = vec![10.0];
        assert_eq!(mean(&numbers), Some(10.0));

        // Test with negative numbers
        let numbers = vec![-1.0, -2.0, -3.0];
        assert_eq!(mean(&numbers), Some(-2.0));
    }

    #[test]
    fn test_median() {
        // Test with a list of odd length
        let mut numbers = vec![1.0, 3.0, 2.0];
        assert_eq!(median(&mut numbers), Some(2.0));

        // Test with a list of even length
        let mut numbers = vec![1.0, 2.0, 3.0, 4.0];
        assert_eq!(median(&mut numbers), Some(2.5));

        // Test with an empty list
        let mut numbers: Vec<f64> = vec![];
        assert_eq!(median(&mut numbers), None);

        // Test with one element
        let mut numbers = vec![10.0];
        assert_eq!(median(&mut numbers), Some(10.0));
    }
}
